### ISSUE TITLE
Type of object keys accept wrong type

### ISSUE THREAD

**Issue by zuhairtaha** [microsoft/TypeScript#50198]
# Bug Report

An issue with object types

### 🔎 Search Terms

Object type key string or number

### 🕗 Version & Regression Information

In the last version of Typescript 4.8 and I guess in all previous versions if you define an object type like this

### ⏯ Playground Link

[Playground link](https://www.typescriptlang.org/play?ts=4.8.0-beta#code/MYewdgzgLgBCBGArAXDA3jA2gawKYE9VoAnASzAHMBdIqMymAXxgF51GBuAKC4UUwBEAMxAgBVVjAHwAhsQHc+mAIwS20mQC8FMAPS6YAFQAWpCDAjGQAVwA2AEwtWA7jBlgYuYsRDEuQA)

### 💻 Code

```ts
const obj: { [key: string]: string } = {};

obj["foo"] = "bar";
obj[1] = "baz"; // This should show an error

```

### 🙁 Actual behavior

It accept the key of number types

### 🙂 Expected behavior

It should reject and show an error

**Comment by MartinJohns** [comment]
Duplicate of https://github.com/microsoft/TypeScript/issues/45446#issuecomment-898672792.
